cmd/wallet-web/src/components/CredentialDetails/RenameCredentialModal.vue

```diff
@@ -0,0 +1,176 @@
+    },
+  },
+  emits: ['close'],
+  setup(props) {
+    const store = useStore();
+    const agentInstance = computed(() => store.getters['agent/getInstance']);
+    const currentUser = computed(() => store.getters.getCurrentUser);
+    const { t } = useI18n();
+    const showModal = ref(false);
+    watch(
+      () => props.show,
+      (show) => {
+        showModal.value = show;
+      }
+    );
+
+    return { agentInstance, currentUser, showModal, t };
+  },
+  data() {
+    return {
+      credentialName: '',
+      loading: false,
+      submitted: false,
+    };
+  },
+  computed: {
+    errorMessage() {
+      if (this.submitted && this.nameEmpty) {
+        return this.t('CredentialDetails.emptyError');
+      } else if (!this.nameValid) {
+        return this.t('CredentialDetails.patternError');
+      } else return '';
+    },
+    // Returns true if the length of the trimmed name is 0
+    nameEmpty() {
+      return this.credentialName.trim().replace(/ +/g, '\\s+').length === 0;
+    },
+    // Returns true if the name only contains letters, numbers or whitespace characters
+    nameValid() {
+      const re = new RegExp('^[a-zA-Z\\d]+(?:[a-zA-Z\\d\\s]+)*$');
+      return re.test(this.credentialName.trim().replace(/  +/g, ' '));
+    },
+    pattern() {
+      let existingNamesRegex = `(?!^${this.credentialName.trim().replace(/ +/g, '\\s+')}$)`;
+      return `${existingNamesRegex}(^[a-zA-Z\\d]+(?:[a-zA-Z\\d\\s]+)*$)`;
+    },
+  },
+  watch: {
+    showModal: function () {
+      if (!this.showModal.value) {
+        this.submitted = false;
+        this.credentialName = '';
+      }
+    },
+  },
+  methods: {
+    updateCredentialName({ name }) {
+      this.credentialName = name;
+      this.submitted = false;
+    },
+    async renameCredential(credentialId, vaultName) {
+      this.submitted = true;
+      if (this.credentialName.length && this.nameValid) {
+        this.loading = true;
+        const { user, token } = this.currentUser.profile;
+        const credentialManager = new CredentialManager({ agent: this.agentInstance, user });
+        const collectionManager = new CollectionManager({ agent: this.agentInstance, user });
+        const { contents } = await collectionManager.getAll(token);
+        this.vaults = Object.values(contents).map((vault) => vault);
+        const vaultID = this.vaults.find((vault) => vault.name === vaultName).id;
+        try {
+          await credentialManager.updateCredentialMetadata(token, credentialId, {
+            name: this.credentialName,
+            collection: vaultID,
+          });
+
+          let metadata = await credentialManager.getCredentialMetadata(token, credentialId);
+          credentialMutations.setCredentialsOutdated(true);
+          this.showModal = false;
+          this.loading = false;
+          this.submitted = false;
+          this.$emit('close');
+        } catch (e) {
+          console.error('Error while renaming credential:', e);
```

Comment:
Can't we use standard error handler/display of system error in this case?

Reply:
can be done as part of #1600

Reply:
Having further discussion with the design team, system errors in the modals should a toast errors like this https://app.abstract.com/projects/9ec92568-60a2-4fee-8215-5aa997d52428/branches/dc10bcee-d9a1-44f9-b644-8f843c6f1a19/commits/a0222a405989d1245b0548d775b3a9aae6fba3ed/files/89ee2d9d-cd69-4357-9433-e0be2f7748fb/layers/EC425024-0893-4172-98B4-77CDB5298D58


This should be separate task #1600  @sudeshrshetty

Reply:
why cant we show it like existing error scenario for this page? just different text (copied from system error page), add a TODO and fix it later once you get something from design team

Reply:
I had discussion with design team and have added this as toast message

Reply:
<img width="1677" alt="Screen Shot 2022-03-22 at 8 27 14 PM" src="https://user-images.githubusercontent.com/7862595/159603954-384f7505-0db9-4010-85a5-2c5b9f2003f8.png">
As per the discussion with the design team i have added toast and have requested team to provide abstract design for the same as well